cscs-checks/prgenv/affinity_check.py

```diff
@@ -23,8 +23,8 @@ def __init__(self, variant):
         self.maintainers = ['RS', 'SK']
         self.tags = {'production', 'scs', 'maintenance', 'craype'}
 
-    @rfm.run_before('compile')
-    def setup_per_partition(self):
+    @rfm.run_after('setup')
+    def prepare_test(self):
```

Comment:
Because of `self.use_multithreading = self.cases[self.variant]['multithreading']` in line 51. I can split it in two functions, the sanity one +
```
    @rfm.run_before('run')
    def set_multithreading(self):
```

Reply:
Better split it two then.